modules/imgcodecs/src/loadsave.cpp

```diff
@@ -464,8 +464,13 @@ imread_( const String& filename, int flags, Mat& mat )
             type = CV_MAKETYPE(CV_MAT_DEPTH(type), 1);
     }
 
-    mat.create( size.height, size.width, type );
-
+    if(allocateMat) mat.create( size.height, size.width, type );
+    else 
+    {
+      CV_Assert(size == mat.size());
+      CV_Assert(type == mat.type());
```

Comment:
I don't see any problem on reallocation of `mat` in case of size/type mismatch. Other OpenCV methods will behave in a same way.

Reply:
Hi @dkurt,
In principle I agree with you that there would be no problem with reallocation.
But the idea of this version of imread is that it uses a given area of memory (perhaps shared).
So in this case I expect that if imread can't use the memory I'm passing, it will give me an error.
Internal reallocation without warning can lead to strange behaviour. 

For example in the code below:

```
Mat img = Mat::zeros(1000, 1000, CV_8UC3);
Mat subImg = img.rowRange(64, 576).colRange(128, 640);
imread(samples::findFile("cv/shared/lena.png"), subImg);
```

the _subImg_ would be reallocated while _img_ remain an empty image.